src/backend/bridge/dml/executor/plan_executor.cpp

```diff
@@ -35,15 +35,124 @@ executor::ExecutorContext *BuildExecutorContext(ParamListInfoData *param_list,
 /*
  * Added for network invoking efficiently
  */
-executor::ExecutorContext *BuildExecutorContext(const std::vector<Value> &params,
-                                                concurrency::Transaction *txn);
+executor::ExecutorContext *BuildExecutorContext(
+    const std::vector<Value> &params, concurrency::Transaction *txn);
 
 executor::AbstractExecutor *BuildExecutorTree(
     executor::AbstractExecutor *root, const planner::AbstractPlan *plan,
     executor::ExecutorContext *executor_context);
 
 void CleanExecutorTree(executor::AbstractExecutor *root);
 
+/*
+ * Execute the subplan and get a value from the result.
+ */
+Value PlanExecutor::ExecutePlanGetValue(const planner::AbstractPlan *plan,
+                                        const std::vector<Value> &params) {
+  peloton_status p_status;
+
+  //  if (plan == nullptr) return p_status;
+
+  LOG_TRACE("PlanExecutor Start ");
+
+  bool status;
+  bool init_failure = false;
+  bool single_statement_txn = false;
+  Value ret_val;
+  auto &txn_manager = concurrency::TransactionManagerFactory::GetInstance();
+  auto txn = peloton::concurrency::current_txn;
+  // This happens for single statement queries in PG
+  if (txn == nullptr) {
+    single_statement_txn = true;
+    txn = txn_manager.BeginTransaction();
+  }
+  assert(txn);
+
+  LOG_TRACE("Txn ID = %lu ", txn->GetTransactionId());
+  LOG_TRACE("Building the executor tree");
+
+  // Use const std::vector<Value> &params to make it more elegant for network
+  std::unique_ptr<executor::ExecutorContext> executor_context(
+      BuildExecutorContext(params, txn));
+  // auto executor_context = BuildExecutorContext(param_list, txn);
+
+  // Build the executor tree
+  std::unique_ptr<executor::AbstractExecutor> executor_tree(
+      BuildExecutorTree(nullptr, plan, executor_context.get()));
+
+  LOG_TRACE("Initializing the executor tree");
+
+  // Initialize the executor tree
+  status = executor_tree->Init();
+
+  // Abort and cleanup
+  if (status == false) {
+    init_failure = true;
+    txn->SetResult(Result::RESULT_FAILURE);
+    goto cleanup;
+  }
+
+  LOG_TRACE("Running the executor tree");
+
+  // Execute the tree until we get result tiles from root node
+  for (;;) {
+    status = executor_tree->Execute();
+
+    // Stop
+    if (status == false) {
+      break;
+    }
+
+    std::unique_ptr<executor::LogicalTile> logical_tile(
+        executor_tree->GetOutput());
+
+    // Some executors don't return logical tiles (e.g., Update).
+    if (logical_tile.get() == nullptr) {
+      continue;
+    }
+
+    // Go over the logical tile
+    for (oid_t tuple_id : *logical_tile) {
+      expression::ContainerTuple<executor::LogicalTile> cur_tuple(
+          logical_tile.get(), tuple_id);
+      ret_val = cur_tuple.GetValue(0);
+      break;
+    }
+  }
+
+  // Set the result
+  p_status.m_processed = executor_context->num_processed;
+//  p_status.m_result_slots = slots;
```

Comment:
Because this method only returns the first value of of result tuple in execution, slots are for the main plan instead of initplan.